src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_DepArrPage.js

```diff
@@ -496,11 +649,29 @@ class CJ4_FMC_DepArrPage {
         fmc.onRightInput[5] = () => {
             if (rsk6Field == "CANCEL MOD>") {
                 fmc.setMsg("Working...");
-                if (fmc.flightPlanManager.getCurrentFlightPlanIndex() === 1) {
+                if (fmc.modVfrRunway == true && fmc.vfrLandingRunway && fmc.flightPlanManager.getCurrentFlightPlanIndex() === 1) {
+                    fmc.eraseTemporaryFlightPlan(() => {
+                        fmc.vfrLandingRunway = undefined;
+                        fmc.modVfrRunway = false;
+                        fmc.fpHasChanged = false;
+                        fmc.setMsg();
+                        CJ4_FMC_DepArrPage.ShowArrivalPage(fmc);
+                    });
+                }
+                else if (fmc.modVfrRunway == true && fmc.vfrLandingRunway == undefined && fmc.flightPlanManager.getCurrentFlightPlanIndex() === 1) {
+                    fmc.eraseTemporaryFlightPlan(() => {
+                        fmc.vfrLandingRunway = fmc.deletedVfrLandingRunway;
```

Comment:
Get rid of this second if
Put something like
`fmc.vfrLandingRunway = (fmc.vfrLandingRunway == undefined) ? fmc.deletedVfrLandingRunway : undefined` in the first one

Reply:
I don't follow. they're different if criteria?